### PR TITLE
docs: split out error handling clause

### PR DESCRIPTION
**Pull request by mvayngrib**
i see people getting this wrong constantly

taken from our best practices in coda

### CODE REVIEW

**Review by timlanahan — APPROVED** [review]
utACK

nice -- this makes it pretty obvious

File: README.md [hunk 1]
```diff
@@ -22,7 +22,14 @@ See points in parentheses before each item. Negative points mean the candidate c
 
 ### Code Quality
 
-- [ ] (2) Code is readable: linted, prettified, and written in a consistent style
+- [ ] (1) Code is readable: linted, prettified, and written in a consistent style
+- [ ] (1) Good error handling
+  - Let errors bubble up unless you know exactly how to handle them.
+  - Avoid swallowing errors in low-level libraries/modules, e.g. in exodus-hydra modules.
```
Comment (mvayngrib) [inline README.md]: non-Exodus people don't know hydra
```suggestion
  - Avoid swallowing errors in low-level libraries/modules.
```

File: README.md [hunk 2]
```diff
@@ -22,7 +22,14 @@ See points in parentheses before each item. Negative points mean the candidate c
...
 ### Code Quality
 
-- [ ] (2) Code is readable: linted, prettified, and written in a consistent style
+- [ ] (1) Code is readable: linted, prettified, and written in a consistent style
+- [ ] (1) Good error handling
+  - Let errors bubble up unless you know exactly how to handle them.
+  - Avoid swallowing errors in low-level libraries/modules.
+  - Don’t return null / undefined to indicate failure.
+  - Avoid giant try/catch statements. It should be clear from the `try/catch` what might throw and why.
+  - See also Error Handling: Avoid Blind catch-es
```
Comment (mvayngrib) [inline README.md]: rm ref to private doc
```suggestion
```

File: README.md [hunk 3]
```diff
@@ -22,7 +22,13 @@ See points in parentheses before each item. Negative points mean the candidate c
...
 ### Code Quality
 
-- [ ] (2) Code is readable: linted, prettified, and written in a consistent style
+- [ ] (1) Code is readable: linted, prettified, and written in a consistent style
+- [ ] (1) Good error handling
+  - Let errors bubble up unless you know exactly how to handle them.
+  - Avoid swallowing errors in low-level libraries/modules.
+  - Don’t return null / undefined to indicate failure.
+  - Avoid giant try/catch statements. It should be clear from the `try/catch` what might throw and why.
+  - Don’t return null / undefined to indicate failure.
```
Comment (andrejborstnik) [inline README.md]: Dup

**Review by andrejborstnik — APPROVED** [review]
utACK

File: README.md (hunk 3 above)
Comment (olistic) [reply README.md]:
```suggestion
  - Don’t return null / undefined to indicate failure.
  - Avoid giant try/catch statements. It should be clear from the `try/catch` what might throw and why.
```

**Review by olistic — APPROVED** [review]

**Pull request merged**